Ticket #3954 Add IsAuthorized field to PatientConsultationInfo.cs

simplecode80 opened the pull request:
Hello Aaron, could you pleas look on this small pull request.

lorddev commented on SnapMD.ConnectedCare.ApiModels/PatientConsultationInfo.cs:
```diff
@@ -24,6 +24,11 @@ public class PatientConsultationInfo
+        /// <summary>
+        /// For patients who depend from they guardians this flag shows whether patient authorized by any of his guardians or not.
+        /// </summary>
+        public bool IsAuthorized { get; set; }
```
@simplecode80 @Number1MSFanboy This should be handled differently. The property does not indicate that the dependent Patient is "authorized", but that the current user is "authorized" to handle the dependent's matters. Therefore it should be something like a "GuardianAuthorizationStatus" and incude a collection of user IDs for users allowed to manage the user's health matters.

simplecode80 replied:
@lorddev I will remove comment if it confused. 
This property necessary for storing value from PatientProfile.IsAuthorized() method for more details see SnapMD.Data.Entities.PatientProfile class. 

This information necessary in Consultation page.

simplecode80 commented:
@lorddev I pushed new new version without comment under IsAuthorized field (like other fields in PatientConsultationInfo class). 

Could you please look again?

lorddev commented:
@simplecode80 IsAuthorized is a property of the FamilyGroup model. It is not an appropriate property of a patient model. See my comments above.

Please change the model as follows:
1. Add a new class
   
   ```
   public class Guardian
   {
       public int UserId { get; set; }
   
       public string Name { get; set; }
   
       public bool IsAuthorized { get; set; }
   }
   ```
2. Add the Guardian as a property of PatientConsultationInfo:
   
   ```
   public Guardian Guardian { get; set; }
   ```
3. Remove `GuardianFirstName`, `Guardianname`, and `GuardianLastName` from the PatientConsultationInfo object.

simplecode80 commented:
@lorddev thanks for explanation, yes you right such approach is better than put all information about Guardians directly in PatientConsultationInfo class. I will check whether information about guardians used somewhere and refactor it.

I think I chose wrong name for new Field, what I really wanted to add is a property which shows whether it possible to appoint a consultation for this patient or not.

### I have one last question about this pull request:

Usually in application we need to know is it possible to appoint consultation to the patient or not. 
This business logic currently stored in SnapMD.Data.Entities.PatientProfile.cs class ,here is code:

```
    //ToDo: Change method name to more correct.
    public bool IsAuthorized() 
    {
        if (IsDependent == "Y")
        {
            // Some code has been removed for brevity.
            return UserDependentRelationAuthorizations.Any(x => x.IsAuthorized == "Y" );
        }

        return true;
    }
```

So when I added "IsAuthorized" field inside PatientConsultationInfo (I used wrong name for property) I wanted to provide information about patient status in Hospital (can we appoint consultation for this patient, etc..) maybe name like "CanAppointConsultaionForPatient" will be better?

I added this field because I did not wanted to duplicate business logic from PatientProfile class in client side.

Could you please clarify is it Okay if our api model contains not only fields from EF entities (FirstName, LastName, etc...) but also fields with business logic like CanAppointConsultaionForPatient field?

Could you please clarify what will be correct approach: 
1. Add field with correct name which provide patient status in hospital (can we appoint consultation for this patient) for example "CanAppointConsultaionForPatient".
2. Add all information (list of Guardians and IsDependent fields from PatientProfiel entity) inside PatientConsultationInfo, in client side create method like one above and check whether we can create appointment for customer or not.

lorddev commented:
@simplecode80 I'm not sure what you mean by "patient status in hospital". This does not correspond with the existing IsAuthorized property of the entity. Can you please explain what you mean by "in hospital"?

simplecode80 commented:
@lorddev Implemented according to your Notes.

Add information about patient authorization.
1. Add 'IsDependent' property from PatientProfile
2. Add information about patient guardians.

Commit:
Ticket #3954 and #3654

Add information about patient authorization.
1. Add 'IsDependent' property from PatientProfile
2. Add information about patient guardians.
